RecoEgamma/EgammaPhotonProducers/src/PhotonProducer.cc

```diff
@@ -210,6 +213,16 @@ void PhotonProducer::produce(edm::Event& theEvent, const edm::EventSetup& theEve
   }
   if( validEcalRecHits) endcapRecHits = *(endcapHitHandle.product());
 
+  bool validPreshowerRecHits=true;
+  Handle<EcalRecHitCollection> preshowerHitHandle;
+  theEvent.getByToken(preshowerHits_, preshowerHitHandle);
+  EcalRecHitCollection preshowerRecHits;
+  if (!preshowerHitHandle.isValid()) {
+    throw cms::Exception("PhotonProducer") 
+      << "Error! Can't get the preshowerEcalHits";
+  }
```

Comment:
please make this a edm::LogError("PhotonProducer") Then set validPreshowerRecHits=false, 

This is what is done for the other ECAL hits : https://github.com/cms-sw/cmssw/blob/master/RecoEgamma/EgammaPhotonProducers/src/PhotonProducer.cc#L191-L200